integrations/sharepoint/src/SharepointSync.ts

```diff
@@ -51,7 +54,7 @@ export class SharepointSync {
     return this.kbInstances.get(kbId)!
   }
 
-  async loadAllDocumentsIntoBotpressKB(): Promise<void> {
+  async loadAllDocumentsIntoBotpressKB(clearedKBs?: Set<string>): Promise<void> {
     // 1 - Fetch all files in this doclib
     const items = await this.sharepointClient.listItems()
```

Comment:
nitpick: if you can split your code using comments like this, you could probably extract some methods out of here.